RecoParticleFlow/PFClusterProducer/plugins/Basic2DGenericPFlowClusterizer.cc

```diff
@@ -166,8 +181,12 @@ growPFClusters(const reco::PFCluster& topo,
 	std::abs(refhit->positionREP().eta()) > 0.34 ) {
       cell_layer *= 100;
     }  
-    const double recHitEnergyNorm = 
-      _recHitEnergyNorms.find(cell_layer)->second; 
+
+    std::pair<std::vector<int>,std::vector<double> > recHitEnergyNormDepthPair = _recHitEnergyNorms.find(cell_layer)->second;
+
+    const std::vector<double> recHitEnergyNormV = recHitEnergyNormDepthPair.second;
+    const std::vector<int> recHitDepthV = recHitEnergyNormDepthPair.first;
```

Comment:
copies of vectors in this code are really bad for the technical performance.
Please switch to using const references.
This is at least a quick solution for this implementation.

I think that it will be more efficient to change back _recHitEnergyNorms  to be just a map of <int,double>.
The meaning of the int will change to be cell_layer+[max cell_layer]*depth (or the other way around depth+[max depth]*cell_layer). The navigation will be more obvious this way.